include/CFinalBoneHierarchy.h

```diff
@@ -101,69 +101,70 @@ namespace scene
 
-				boneFlatArray = (BoneReferenceData*)malloc(_blob->calcBonesByteSize(this));
-				boneTreeLevelEnd = (size_t*)malloc(_blob->calcLevelsByteSize(this));
-				keyframes = (float*)malloc(_blob->calcKeyFramesByteSize(this));
-				interpolatedAnimations = (AnimationKeyData*)malloc(_blob->calcInterpolatedAnimsByteSize(this));
-				nonInterpolatedAnimations = (AnimationKeyData*)malloc(_blob->calcNonInterpolatedAnimsByteSize(this));
 				boneNames = new core::stringc[boneCount];
+				boneFlatArray = (BoneReferenceData*)malloc(sizeof(BoneReferenceData)*boneCount);
+				boneTreeLevelEnd = (size_t*)malloc(sizeof(size_t)*NumLevelsInHierarchy);
+				keyframes = (float*)malloc(sizeof(float)*keyframeCount);
+				interpolatedAnimations = (AnimationKeyData*)malloc(sizeof(AnimationKeyData)*keyframeCount);
+				nonInterpolatedAnimations = (AnimationKeyData*)malloc(sizeof(AnimationKeyData)*keyframeCount);
 
-				memcpy(boneFlatArray, blobData + _blob->calcBonesOffset(this), _blob->calcBonesByteSize(this));
-				memcpy(boneTreeLevelEnd, blobData + _blob->calcLevelsOffset(this), _blob->calcLevelsByteSize(this));
-				memcpy(keyframes, blobData + _blob->calcKeyFramesOffset(this), _blob->calcKeyFramesByteSize(this));
-				memcpy(interpolatedAnimations, blobData + _blob->calcInterpolatedAnimsOffset(this), _blob->calcInterpolatedAnimsByteSize(this));
-				memcpy(nonInterpolatedAnimations, blobData + _blob->calcNonInterpolatedAnimsOffset(this), _blob->calcNonInterpolatedAnimsByteSize(this));
-				const char* strings = (const char*)(blobData + _blob->calcBoneNamesOffset(this));
-				for (int i = 0; i < boneCount; ++i)
-				{
-					boneNames[i] = core::stringc(strings);
-					strings += strlen(strings)+1;
-				}
+				for (size_t i = 0; i < boneCount; ++i)
+					boneNames[i] = _boneNamesBegin[i];
+				memcpy(boneFlatArray, _bonesBegin, sizeof(BoneReferenceData)*boneCount);
+				memcpy(boneTreeLevelEnd, _levelsBegin, sizeof(size_t)*NumLevelsInHierarchy);
+				memcpy(keyframes, _keyframesBegin, sizeof(float)*keyframeCount);
+				memcpy(interpolatedAnimations, _interpAnimsBegin, sizeof(AnimationKeyData)*keyframeCount);
+				memcpy(interpolatedAnimations, _nonInterpAnimsBegin, sizeof(AnimationKeyData)*keyframeCount);
 			}
 
-			//! Function filling blob's storage with the object's data. Meant to be used by loaders only.
+			//! Function filling blob's storage with the object's data. Meant to be used by exporters only.
 			/**
 			@param _dataPtr Pointer to pre-allocated memory (heap or stack - doesn't matter) of size sufficient to hold all object's data. This size can be calculated with core::FinalBoneHierarchyBlobV1::calcBlobSizeForObj().
-			@param[out] _size Through this parameter size (in bytes) of written data is returned.
 			@see @ref CBAWMeshFileLoader CBAWMeshWriter
 			*/
-			void fillExportBlob(void* _dataPtr, uint32_t* _size)
+			size_t serializeToBlob(void* _dataPtr) const
```

Comment:
can we make this a template function, so that the blob is filled inside a Blob member function?
`template <typename B>
B* serializeToBlob(void* _stackPtr=NULL, const size_t& _size=0)
{
    return VariableSizeBlob<B,CFinalBoneHierarchy>::createAndTryOnStack(this,_stackPtr,_size);
}`

And then make the 'VariableSizeBlob::createAndTryOnStack' function actually use placement new and the proper private constructor (which fills all data) to have a unified API